drivers/sensor/adxrs290/adxrs290.c

```diff
@@ -0,0 +1,532 @@
+	if (ret) {
+		return ret;
+	}
+
+	data &= ~ADXRS290_DATA_READY_INT_MASK;
+
+	if (activate) {
+		data |=  0x01;
+	}
+
+	ret = adxrs290_write_reg(dev, ADXRS290_DATA_READY_REG, data);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int adxrs290_set_low_pass_filter(struct device *dev, int lf_pole)
+{
+	u8_t data;
+	int ret;
+
+	ret = adxrs290_read_reg(dev, ADXRS290_BANDPASS_FILTER, &data);
+	if (ret) {
+		return ret;
+	}
+
+	data &= ~ADXRS290_BPF_LPF_MASK;
+	data |= (lf_pole & ADXRS290_BPF_LPF_MASK);
+
+	ret = adxrs290_write_reg(dev, ADXRS290_BANDPASS_FILTER, data);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int adxrs290_set_high_pass_filter(struct device *dev, int hf_pole)
+{
+	u8_t data;
+	int ret;
+
+	ret = adxrs290_read_reg(dev, ADXRS290_BANDPASS_FILTER, &data);
+	if (ret) {
+		return ret;
+	}
+
+	data &= ~ADXRS290_BPF_HPF_MASK;
+	data |= ((hf_pole << ADXRS290_BPF_HPF_OFFSET) &
+		 ADXRS290_BPF_HPF_MASK);
+
+	ret = adxrs290_write_reg(dev, ADXRS290_BANDPASS_FILTER, data);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int adxrs290_temp_sensor_enable(struct device *dev, u8_t enable)
+{
+	u8_t data;
+	int ret;
+
+	ret = adxrs290_read_reg(dev, ADXRS290_POW_CTRL_REG, &data);
+	if (ret) {
+		return ret;
+	}
+
+	data &= ~ADXRS290_POW_CTRL_TEMP_EN_MASK;
+
+	if (enable) {
+		data |=  ADXRS290_POW_CTRL_TEMP_EN_MASK;
+	}
+
+	ret = adxrs290_write_reg(dev, ADXRS290_POW_CTRL_REG, data);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int adxrs290_read_xy(struct device *dev, s16_t *x, s16_t *y)
+{
+	u8_t buf[4];
+	int ret;
+
+	if (adxrs290_data.standby) {
+		ret = adxrs290_standby_read_xy(dev, x, y);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	ret = adxrs290_read_regs(dev, ADXRS290_GYR_X_L, buf, 4);
+	if (ret) {
+		return ret;
+	}
+
+	*x = buf[2] | (buf[3] << 8);
+	*y = buf[0] | (buf[1] << 8);
+
+	return 0;
+}
+
+static int adxrs290_read_temperature(struct device *dev, s16_t *temp)
+{
+	u8_t buf[2];
+	int ret;
+
+	if (adxrs290_data.standby) {
+		return adxrs290_standby_read_temperature(dev, temp);
+	}
+
+	ret = adxrs290_read_regs(dev, ADXRS290_TEMP_L, buf, 2);
+	if (ret) {
+		return ret;
+	}
+
+	*temp = buf[0] | ((0x0F & buf[1]) << 8);
+
+	return 0;
+}
+
+static int adxrs290_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct adxrs290_data *data = dev->driver_data;
+	int ret;
+
+	ret = adxrs290_read_xy(dev, &data->x, &data->y);
+	if (ret) {
+		return ret;
+	}
+
+	ret = adxrs290_read_temperature(dev, &data->temp);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int adxrs290_channel_get(struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct adxrs290_data *data = dev->driver_data;
+
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_X:
+		/* Angular velocity around the X axis, in radians/s. */
+		val->val1 = (s32_t)data->x / ADXRS290_GYR_SCALE_FACT;
+		val->val2 = ((s32_t)data->x % ADXRS290_GYR_SCALE_FACT)
```

Comment:
You still need to divide val2 with ADXRS290_GYR_SCALE_FACT. This applies to GYRO_Y and TEMP as well.